src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java

```diff
@@ -31,30 +35,38 @@ public TownyContext(@NonNull final CarbonChat carbonChat) {
 
       return true;
     });
-  }
 
-  @EventHandler(ignoreCancelled = true)
-  public void onChannelSwitch(final ChannelSwitchEvent event) {
-    final Object town = event.channel().context(KEY);
+    final TownyContext instance = this;
```

Comment:
Why?

Reply:
Checkstyle - it complains about the omission of `this` which isn't possible due to the anonymous class.
After some trial and error I found this to be the easiest way to satisfy it.

Reply:
You can use TownyContext.this.<whatever> instead

Reply:
4ad0434